Update ANGLE to 2025-04-25 (34d30914048ac98a6f50f84aa630cc285b75bc79)

kkinnunen-apple opened the pull request:
#### cf222c5b83fc18096e0b17dadf1270d1aee7a9bc
<pre>
Update ANGLE to 2025-04-25 (34d30914048ac98a6f50f84aa630cc285b75bc79)
<a href="https://bugs.webkit.org/show_bug.cgi?id=292054">https://bugs.webkit.org/show_bug.cgi?id=292054</a>
<a href="https://rdar.apple.com/problem/150026170">rdar://problem/150026170</a>

Reviewed by Mike Wyrzykowski

Contains upstream commits:
git log --oneline 1c0bcb06e36fe6920fb09512bcfe7519fe1bcbb2..34d30914048ac98a6f50f84aa630cc285b75bc79 --pretty=%h %s
34d3091404 Roll Chromium from 66b5465c817a to 85dd9668972f (684 revisions)
ef61ff8db5 Improve the judgement of texture and attachment completeness
4429afe9e8 Tests: Add Black Beacon trace
d08e9e9c9c Fix use of OpenGL ES 2.0 commands in ES 1.0 tests
272ffbdacb Enhance entry point validation autogen scripts
ca9597674d Include IOKit.framework with `is_mac || angle_enable_cgl`
e5680ba624 Fix mInFlightCommands.empty() assert in CommandQueue destroy
ff110c8663 Early-return instead of rebinding the same subject
88710dc139 Bumping astc-encoder version to 5.2.0 (latest release)
be0788b91c Roll vulkan-deps from 2e021a754b60 to ac1b0b870946 (15 revisions)
1e4ad682f1 Roll Chromium from 90de43ba04db to 66b5465c817a (568 revisions)
4c1c7c083f Include entry point names in error messages
56b62c625a Include entry point names in context lost error messages
d392eeaea9 Vulkan: Force host-image-copy for LUMA textures on ARM
cdcf6a6348 Add R10X6G10X6B10X6A10X6 support on angle
555038b8ae Translator Fuzzer: Enable required options
d698500d38 Fix an ASSERT() crash in ExternalFence SyncWaitFd().
691140da28 Vulkan: Do not skip VUID-vkQueueSubmit-pSignalSemaphores-00067
35c550a49d Roll vulkan-deps from c70bb7aae968 to 2e021a754b60 (11 revisions)
6f80d0817f Roll Chromium from fda9025645de to 90de43ba04db (622 revisions)
6f9f343a24 Reland &quot;Vulkan: Prune superseded updates during host-image-copy&quot;
3c554454bb Incorrect BUFFER_DATA_SIZE and UNIFORM_OFFSET for ACB array of array
abe3c63a57 Add CubeMapBug2 test to trigger Angle bug in validation
61bfe86cc4 Autogen context version checks for 1.0 entry points
3f88a50562 GenerateMipmap encode and decode sRGB when GL_SKIP_DECODE_EXT is set
50a4ad195a Revert &quot;Vulkan: Prune superseded updates during host-image-copy&quot;
69a0fc24c2 Use dedicated error messages for context version checks
db12f43b82 Autogen context version checks for 3.0 entry points
12fcd01df1 Roll vulkan-deps from 61080f55d11e to c70bb7aae968 (17 revisions)
5bc78fca73 Roll VK-GL-CTS from 76c1572eaba4 to dd7e71367795 (10 revisions)
3c30881b47 Roll Chromium from ea07a56dbf64 to fda9025645de (411 revisions)
81f1a733f4 Fix use of OpenGL ES 3.0 commands in image tests
962e147a59 Roll vulkan-deps from 5880b384399b to 61080f55d11e (17 revisions)
b0fc13f853 Roll Chromium from a3c80076ee75 to ea07a56dbf64 (376 revisions)
f355e2b37e Vulkan: Remove preferDriverUniformOverSpecConst
07eeca121d Metal: prevent padding in state cache&apos;s structs.
f6a350de6d Metal: fix rasterSamplecount not used in comparing render passes.
d4336f18b0 Manual roll Chromium from 5d2f55eb5444 to a3c80076ee75 (831 revisions)
f8a8adf63c Vulkan: Prune superseded updates during host-image-copy
a9f2f87f9d Vulkan: Update version macros
151a9ed2c6 Roll SwiftShader from 2d70dae3d0ca to 0e75e7030ccb (1 revision)
4928b7f720 Roll vulkan-deps from b552240b4d50 to 5880b384399b (7 revisions)
4b9525b266 Remove all getClientMajorVersion helpers
9db4bf6e58 Autogen context version checks for 3.1 &amp; 3.2 entry points
b86d7d8f57 Android: Add Galaxy S24 support (Xclipse)
760562b5b8 Roll vulkan-deps from f286d9d385ce to b552240b4d50 (7 revisions)
65011819da Roll Chromium from f2e4beff997e to 5d2f55eb5444 (608 revisions)
53279e9294 Traces: Update traces that need ES 3.2
ccabbbca67 Suppress flaky test on Nvidia/D3D
a6a7de5aec Vulkan: Sync perf counters in finishImpl(...)
408f58532b Make gl::Version data members private
b64d380ec5 Vulkan: Ghost texture&apos;s image backing if overwritten
09a81a6836 Roll third_party/libpng/src/ d2ece84bd..8cc222cd3 (277 commits)
dbfbec0001 Roll vulkan-deps from 72ecfe9950d2 to f286d9d385ce (9 revisions)
f2a7b27614 Roll Chromium from f054743ee574 to f2e4beff997e (597 revisions)
0ed51ec79a Vulkan: Disable color attachment usage for LUMA formats
ca0457f688 Vulkan: Fix parsed QCOM major version
49d0a332be Vulkan: Remove ring buffer allocators
cf7eeec271 Simplify client version compare operations
838c6596e8 Add EGL image tests where source texture is modified
459694920e Add Mac ARM64 builders
33e035f96e Enforce correct usage of gl::Version
e39bcdc310 Support GLES query counter bits
cee22c4392 Roll vulkan-deps from a24af46b3d3b to 72ecfe9950d2 (10 revisions)
4be3143c90 Vulkan: Vertex attribute hole crash workaround
ac8607a92e Vulkan: Disable graphics pipeline on ARM due to performance loss
08e7dfffc6 Roll SwiftShader from 2b323370501c to 2d70dae3d0ca (2 revisions)
dbb0eb69b8 Roll Chromium from 95d3d27a122a to f054743ee574 (545 revisions)
a3f2545f6b Tests that exercise overwriting textures while in use
d32c6c030b Vulkan: Treat present SURFACE_LOST_KHR as OUT_OF_DATE
a5d750fb72 Add test to check EGL_KHR_platform_android
5dd0557841 Revert &quot;GL: Allow untranslated shaders to pass through on GLES&quot;
9554950cae Roll Chromium from 75621a1b9953 to 95d3d27a122a (1865 revisions)
e868696a9f Roll vulkan-deps from 5ec40ef70a52 to a24af46b3d3b (14 revisions)
1664d99098 Roll VK-GL-CTS from ad6225aacbc5 to 76c1572eaba4 (12 revisions)
9662829655 Fix GL_OES_texture_3D entry points validation
4e77552b86 GL: Allow untranslated shaders to pass through on GLES
d5ab13318a Reland^2 &quot;Enable VVL on vulkan instance created in GetSystemInfoVulkanWithICD&quot;
0353ed2ec1 Roll vulkan-deps from 792a8d875adc to 5ec40ef70a52 (3 revisions)
78048112db Fix IndexRange::vertexIndexCount calculation

Canonical link: <a href="https://commits.webkit.org/294185@main">https://commits.webkit.org/294185@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/1bd6694a97aff61c4136aede38abcdc70fbba1a7

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/101061 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/20723 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/11026 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/106207 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/51687 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/103102 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/21032 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/29217 "Built successfully") | [❌ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/76975 "Found 1 new test failure: fast/canvas/webgl/simulated-vertexAttrib0-invalid-indicies.html (failure)") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/34004 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/104068 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/16196 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/91284 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/57323 "Passed tests") | 
| [✅ 🧪 webkitpy](https://ews-build.webkit.org/#/builders/6/builds/100535 "Passed tests") | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/16011 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/9310 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/51035 "Built successfully") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/93728 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/85917 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/9367 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/108563 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/99670 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/28189 "Built successfully") | [❌ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/20755 "Found 2 new test failures: http/tests/iframe-monitor/data-url-resource.html imported/w3c/web-platform-tests/css/css-view-transitions/transition-in-hidden-page.html (failure)") | [❌ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/85943 "Found 1 new test failure: imported/w3c/web-platform-tests/css/css-view-transitions/active-view-transition-type-on-non-root.html (failure)") | 
| [✅ 🧪 services](https://ews-build.webkit.org/#/builders/28/builds/100619 "Passed tests") | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/28551 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/87484 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/85481 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/21757 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/30201 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/7928 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/22276 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/28119 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/33387 "Built successfully") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/123295 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/27931 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/34348 "Passed tests") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/31251 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/29489 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

mwyrzykowski reviewed (approved):
Looks like this is an import, r+ given that

kkinnunen-apple committed:
Update ANGLE to 2025-04-25 (34d30914048ac98a6f50f84aa630cc285b75bc79)
https://bugs.webkit.org/show_bug.cgi?id=292054
rdar://problem/150026170

Reviewed by Mike Wyrzykowski

Contains upstream commits:
git log --oneline 1c0bcb06e36fe6920fb09512bcfe7519fe1bcbb2..34d30914048ac98a6f50f84aa630cc285b75bc79 --pretty=%h %s
34d3091404 Roll Chromium from 66b5465c817a to 85dd9668972f (684 revisions)
ef61ff8db5 Improve the judgement of texture and attachment completeness
4429afe9e8 Tests: Add Black Beacon trace
d08e9e9c9c Fix use of OpenGL ES 2.0 commands in ES 1.0 tests
272ffbdacb Enhance entry point validation autogen scripts
ca9597674d Include IOKit.framework with `is_mac || angle_enable_cgl`
e5680ba624 Fix mInFlightCommands.empty() assert in CommandQueue destroy
ff110c8663 Early-return instead of rebinding the same subject
88710dc139 Bumping astc-encoder version to 5.2.0 (latest release)
be0788b91c Roll vulkan-deps from 2e021a754b60 to ac1b0b870946 (15 revisions)
1e4ad682f1 Roll Chromium from 90de43ba04db to 66b5465c817a (568 revisions)
4c1c7c083f Include entry point names in error messages
56b62c625a Include entry point names in context lost error messages
d392eeaea9 Vulkan: Force host-image-copy for LUMA textures on ARM
cdcf6a6348 Add R10X6G10X6B10X6A10X6 support on angle
555038b8ae Translator Fuzzer: Enable required options
d698500d38 Fix an ASSERT() crash in ExternalFence SyncWaitFd().
691140da28 Vulkan: Do not skip VUID-vkQueueSubmit-pSignalSemaphores-00067
35c550a49d Roll vulkan-deps from c70bb7aae968 to 2e021a754b60 (11 revisions)
6f80d0817f Roll Chromium from fda9025645de to 90de43ba04db (622 revisions)
6f9f343a24 Reland "Vulkan: Prune superseded updates during host-image-copy"
3c554454bb Incorrect BUFFER_DATA_SIZE and UNIFORM_OFFSET for ACB array of array
abe3c63a57 Add CubeMapBug2 test to trigger Angle bug in validation
61bfe86cc4 Autogen context version checks for 1.0 entry points
3f88a50562 GenerateMipmap encode and decode sRGB when GL_SKIP_DECODE_EXT is set
50a4ad195a Revert "Vulkan: Prune superseded updates during host-image-copy"
69a0fc24c2 Use dedicated error messages for context version checks
db12f43b82 Autogen context version checks for 3.0 entry points
12fcd01df1 Roll vulkan-deps from 61080f55d11e to c70bb7aae968 (17 revisions)
5bc78fca73 Roll VK-GL-CTS from 76c1572eaba4 to dd7e71367795 (10 revisions)
3c30881b47 Roll Chromium from ea07a56dbf64 to fda9025645de (411 revisions)
81f1a733f4 Fix use of OpenGL ES 3.0 commands in image tests
962e147a59 Roll vulkan-deps from 5880b384399b to 61080f55d11e (17 revisions)
b0fc13f853 Roll Chromium from a3c80076ee75 to ea07a56dbf64 (376 revisions)
f355e2b37e Vulkan: Remove preferDriverUniformOverSpecConst
07eeca121d Metal: prevent padding in state cache's structs.
f6a350de6d Metal: fix rasterSamplecount not used in comparing render passes.
d4336f18b0 Manual roll Chromium from 5d2f55eb5444 to a3c80076ee75 (831 revisions)
f8a8adf63c Vulkan: Prune superseded updates during host-image-copy
a9f2f87f9d Vulkan: Update version macros
151a9ed2c6 Roll SwiftShader from 2d70dae3d0ca to 0e75e7030ccb (1 revision)
4928b7f720 Roll vulkan-deps from b552240b4d50 to 5880b384399b (7 revisions)
4b9525b266 Remove all getClientMajorVersion helpers
9db4bf6e58 Autogen context version checks for 3.1 & 3.2 entry points
b86d7d8f57 Android: Add Galaxy S24 support (Xclipse)
760562b5b8 Roll vulkan-deps from f286d9d385ce to b552240b4d50 (7 revisions)
65011819da Roll Chromium from f2e4beff997e to 5d2f55eb5444 (608 revisions)
53279e9294 Traces: Update traces that need ES 3.2
ccabbbca67 Suppress flaky test on Nvidia/D3D
a6a7de5aec Vulkan: Sync perf counters in finishImpl(...)
408f58532b Make gl::Version data members private
b64d380ec5 Vulkan: Ghost texture's image backing if overwritten
09a81a6836 Roll third_party/libpng/src/ d2ece84bd..8cc222cd3 (277 commits)
dbfbec0001 Roll vulkan-deps from 72ecfe9950d2 to f286d9d385ce (9 revisions)
f2a7b27614 Roll Chromium from f054743ee574 to f2e4beff997e (597 revisions)
0ed51ec79a Vulkan: Disable color attachment usage for LUMA formats
ca0457f688 Vulkan: Fix parsed QCOM major version
49d0a332be Vulkan: Remove ring buffer allocators
cf7eeec271 Simplify client version compare operations
838c6596e8 Add EGL image tests where source texture is modified
459694920e Add Mac ARM64 builders
33e035f96e Enforce correct usage of gl::Version
e39bcdc310 Support GLES query counter bits
cee22c4392 Roll vulkan-deps from a24af46b3d3b to 72ecfe9950d2 (10 revisions)
4be3143c90 Vulkan: Vertex attribute hole crash workaround
ac8607a92e Vulkan: Disable graphics pipeline on ARM due to performance loss
08e7dfffc6 Roll SwiftShader from 2b323370501c to 2d70dae3d0ca (2 revisions)
dbb0eb69b8 Roll Chromium from 95d3d27a122a to f054743ee574 (545 revisions)
a3f2545f6b Tests that exercise overwriting textures while in use
d32c6c030b Vulkan: Treat present SURFACE_LOST_KHR as OUT_OF_DATE
a5d750fb72 Add test to check EGL_KHR_platform_android
5dd0557841 Revert "GL: Allow untranslated shaders to pass through on GLES"
9554950cae Roll Chromium from 75621a1b9953 to 95d3d27a122a (1865 revisions)
e868696a9f Roll vulkan-deps from 5ec40ef70a52 to a24af46b3d3b (14 revisions)
1664d99098 Roll VK-GL-CTS from ad6225aacbc5 to 76c1572eaba4 (12 revisions)
9662829655 Fix GL_OES_texture_3D entry points validation
4e77552b86 GL: Allow untranslated shaders to pass through on GLES
d5ab13318a Reland^2 "Enable VVL on vulkan instance created in GetSystemInfoVulkanWithICD"
0353ed2ec1 Roll vulkan-deps from 792a8d875adc to 5ec40ef70a52 (3 revisions)
78048112db Fix IndexRange::vertexIndexCount calculation

Canonical link: https://commits.webkit.org/294185@main
